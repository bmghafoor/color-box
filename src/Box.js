import React from "react";

const Box = ({ backgroundColor, height, width }) => {
  const boxStyle = {
    backgroundColor,
    height: `${height}px`,
    width: `${width}px`,
    border: "1px solid black",
  };

  const buttonStyle = {
    display: "inline-block",
    cursor: "pointer",
  };

  const handleDelete = (e) => {
    const parentElement = e.currentTarget.parentNode;
    parentElement.remove();
  };

  return (
    <>
      <div style={boxStyle}>
        <button style={buttonStyle} onClick={handleDelete}>
          X
        </button>
      </div>
    </>
  );
};

export default Box;
