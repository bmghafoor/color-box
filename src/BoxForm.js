import React, { useState } from "react";

const BoxForm = () => {
  const initialState = { backgroundColor: "", height: "", width: "" };
  const [boxData, setBoxData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setBoxData((boxData) => ({ ...boxData, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
  };
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="backgroundColor">Color</label>
      <input
        type="text"
        name="backgroundColor"
        id="backgroundColor"
        placeholder="Enter Color"
        value={boxData.backgroundColor}
        onChange={handleChange}
      />

      <label htmlFor="height">Height</label>
      <input
        type="text"
        name="height"
        id="height"
        placeholder="Enter Height in px"
        value={boxData.height}
        onChange={handleChange}
      />

      <label htmlFor="width">Width</label>
      <input
        type="text"
        name="width"
        id="width"
        placeholder="Enter Width in px"
        value={boxData.width}
        onChange={handleChange}
      />
    </form>
  );
};

export default BoxForm;
