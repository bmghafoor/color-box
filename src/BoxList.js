import React, { useState } from "react";
import Box from "./Box";
import BoxForm from "./BoxForm";

const BoxList = () => {
  const [boxList, setBoxList] = useState([]);

  const addBox = (newBox) => {
    setBoxList((boxList) => [...boxList, newBox]);
  };

  return (
    <div>
      <h3>Boxes</h3>
      <BoxForm addBox={addBox} />
      <div>
        <h4>Here are the boxes</h4>
        {boxList.map(({ backgroundColor, height, width }) => (
          <Box
            backgroundColor={backgroundColor}
            height={height}
            width={width}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
