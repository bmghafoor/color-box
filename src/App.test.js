import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";
import BoxList from "./BoxList";
import BoxForm from "./BoxForm";

it("renders without crashing", function () {
  render(<Box />);
  render(<BoxList />);
  render(<BoxForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
