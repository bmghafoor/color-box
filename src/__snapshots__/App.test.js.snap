// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="border: 1px solid black;"
  >
    <button
      style="display: inline-block; cursor: pointer;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <h3>
      Boxes
    </h3>
    <form>
      <label
        for="backgroundColor"
      >
        Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Enter Color"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Enter Height in px"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter Width in px"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div>
      <h4>
        Here are the boxes
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 3`] = `
<DocumentFragment>
  <form>
    <label
      for="backgroundColor"
    >
      Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Enter Color"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="Enter Height in px"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="Enter Width in px"
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
